Improve support for NewPM

nikic opened the pull request:
This adds various missing bits of support for NewPM and allows us to successfully run stage 2 tests with NewPM enabled.

This does not yet enable NewPM by default, as there are still known issue on LLVM 12 (such as a weak fat LTO pipeline). The plan is to make the switch after we update to LLVM 13.

nikic committed:
Make -Z new-llvm-pass-manager an Option<bool>

To allow it to have an LLVM version dependent default.

nikic committed:
Drop -opt-bisect-limit=0 flag from test

This causes an assertion failure under NewPM, because it also ends
up disabling the NameAnonGlobals pass.

Instead pass -Copt-level=0 to disable optimizations. If that should
be insufficient, we can use -C no-prepopulate-passes.

nikic committed:
Explicitly register instrprof pass

Don't use "passes" for this purpose, explicitly insert it into
the correct place in the pipeline instead.

nikic committed:
Don't force -O1 with ThinLTO

This doesn't seem to be necessary anymore, although I don't know
at which point or why that changed.

Forcing -O1 makes some tests fail under NewPM, because NewPM also
performs inlining at -O1, so it ends up performing much more
optimization in practice than before.

nikic committed:
Explicitly register GCOV profiling pass as well

nikic committed:
Support -C passes in NewPM

And report an error if parsing the additional pass pipeline fails.
Threading through the error accounts for most of the changes here.

rust-highfive commented:
r? @estebank

(rust-highfive has picked a reviewer for you, use r? to override)

nikic commented:
@bors try @rust-timer queue

rust-timer commented:
Awaiting bors try build completion.

@rustbot label: +S-waiting-on-perf

bors commented:
:hourglass: Trying commit 981744e31838ae78a243bbf928fcd1a876a3864a with merge b1b723b5dc62feaeb9a176d276f190f0aa045d0a...
<!-- homu: {"type":"TryBuildStarted","head_sha":"981744e31838ae78a243bbf928fcd1a876a3864a","merge_sha":"b1b723b5dc62feaeb9a176d276f190f0aa045d0a"} -->

bors commented:
:sunny: Try build successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/2271218115)
Build commit: b1b723b5dc62feaeb9a176d276f190f0aa045d0a (`b1b723b5dc62feaeb9a176d276f190f0aa045d0a`)
<!-- homu: {"type":"TryBuildCompleted","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/runs/2271218115"},"merge_sha":"b1b723b5dc62feaeb9a176d276f190f0aa045d0a"} -->

rust-timer commented:
Queued b1b723b5dc62feaeb9a176d276f190f0aa045d0a with parent 39eee173fbcc21462d255b364b87715cd33b62db, future [comparison URL](https://perf.rust-lang.org/compare.html?start=39eee173fbcc21462d255b364b87715cd33b62db&end=b1b723b5dc62feaeb9a176d276f190f0aa045d0a).

rust-timer commented:
Finished benchmarking try commit (b1b723b5dc62feaeb9a176d276f190f0aa045d0a): [comparison url](https://perf.rust-lang.org/compare.html?start=39eee173fbcc21462d255b364b87715cd33b62db&end=b1b723b5dc62feaeb9a176d276f190f0aa045d0a).

Benchmarking this pull request likely means that it is perf-sensitive, so we're automatically marking it as not fit for rolling up. Please note that if the perf results are neutral, you should likely undo the rollup=never given below by specifying `rollup-` to bors.

Importantly, though, if the results of this run are non-neutral **do not** roll this PR up -- it will mask other regressions or improvements in the roll up.

@bors rollup=never
@rustbot label: +S-waiting-on-review -S-waiting-on-perf

Mark-Simulacrum commented:
Very nice results over all, but some pretty major regressions on the self-bootstrap tests:

![image](https://user-images.githubusercontent.com/5047365/113611267-4a9d3700-961c-11eb-86fa-0f9c142dea2e.png)

the8472 commented:
The total bootstrap time only increased by 2.4% though. So multi-core utilization must have improved.

Mark-Simulacrum commented:
No? We run with -j1, so the bootstrap should be single threaded. I'm not sure, but I seem to recall the total being computed client side in the UI - would be worth checking that, I suppose.

the8472 commented:
Nevermind, I eyeballed it wrong. I checked with a spreadsheet now and everything is consistent.

estebank commented:
Some really nice improvements! The code looks fine, but I am not the most qualified person for this part of the codebase. @rust-lang/compiler anyone can take on this review?

nagisa commented:
r? @nagisa

nagisa commented on compiler/rustc_codegen_llvm/src/back/write.rs:
```diff
@@ -472,10 +471,18 @@ pub(crate) unsafe fn optimize_with_new_llvm_pass_manager(
     );
+
+    if let Some(err) = llvm::last_error() {
+        cgcx.create_diag_handler().warn(&format!("Invalid -Cpasses: {}", err));
```
Accept a `diag_handler` in this function like `optimize` does?

nagisa replied:
Additionally, this isn't obvious to me that the error from the call above will always be about `-Cpasses`. It would make more sense to me to do this formatting on the cpp side so that this code is more future-proof.

nagisa commented on compiler/rustc_llvm/llvm-wrapper/PassWrapper.cpp:
```diff
@@ -1045,6 +1066,13 @@ LLVMRustOptimizeWithNewPassManager(
+  if (ExtraPassesLen) {
+    if (auto Err = PB.parsePassPipeline(MPM, StringRef(ExtraPasses, ExtraPassesLen))) {
+      std::string ErrMsg = toString(std::move(Err));
+      LLVMRustSetLastError(ErrMsg.c_str());
```
Should return early here.

nikic replied:
I initially only emitted a warning, but on further consideration it makes sense to error and abort if invalid passes are specified. This required a few more changes to propagate the error.

nagisa reviewed (approved):
Overall this LGTM implementation wise. A couple comments inline. r=me after those are resolved.

---

I wish we had a better explanation for why we could drop some of the older workarounds (e.g. forcing of `-O1`) and an explanation of how things break with `-opt-bisect-limit=0` in the commit message(s).

nikic commented:
> I wish we had a better explanation for why we could drop some of the older workarounds (e.g. forcing of `-O1`) and an explanation of how things break with `-opt-bisect-limit=0` in the commit message(s).

I've made those commit messages a bit more detailed. For `-opt-bisect-limit=0` the issue is an assertion failure because NameAnonGlobals ends up being skipped as well. That might be an LLVM bug, in that the pass should probably be marked as required.

For `-O1` the issue is that NewPM performs inlining at that level, which breaks tests that assume `-O0 -lto=thin` does essentially no optimization. Unfortunately I don't know why it is fine to drop the special handling now -- I'm assuming it *is* fine because no tests break if we do.

nagisa commented on compiler/rustc_codegen_llvm/src/back/write.rs:
```diff
@@ -472,10 +472,15 @@ pub(crate) unsafe fn optimize_with_new_llvm_pass_manager(
+        extra_passes.as_ptr().cast(),
+        extra_passes.len(),
     );
+    result.into_result().map_err(|()| llvm_err(diag_handler, "Failed to run LLVM passes"))
```
I think this `Failed` wants to be non-capitalized. Ends up as `error: Failed…` now which looks out of place.

nagisa commented:
r=me with non-capital f.

nikic commented:
@bors r=nagisa rollup=never

bors commented:
:pushpin: Commit 4903c075fdb6de4b0eba1461de6dae4a2b1b6d0a has been approved by `nagisa`

<!-- @bors r=nagisa 4903c075fdb6de4b0eba1461de6dae4a2b1b6d0a -->
<!-- homu: {"type":"Approved","sha":"4903c075fdb6de4b0eba1461de6dae4a2b1b6d0a","approver":"nagisa"} -->

bors commented:
:hourglass: Testing commit 4903c075fdb6de4b0eba1461de6dae4a2b1b6d0a with merge 07f00298b1780075a702e0d3926cf0fd2aec8eed...
<!-- homu: {"type":"BuildStarted","head_sha":"4903c075fdb6de4b0eba1461de6dae4a2b1b6d0a","merge_sha":"07f00298b1780075a702e0d3926cf0fd2aec8eed"} -->

rust-log-analyzer commented:
The job **`test-various`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/2314463801) [(plain)](https://github.com/rust-lang-ci/rust/commit/07f00298b1780075a702e0d3926cf0fd2aec8eed/checks/2314463801/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
status: exit status: 2
command: "make"
stdout:
------------------------------------------
LD_LIBRARY_PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small:/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib:/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-bootstrap-tools/x86_64-unknown-linux-gnu/release/deps:/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/lib" '/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc' --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small -L /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small  foo.rs -C lto -O --target wasm32-unknown-unknown --cfg a
wc -c < /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small/foo.wasm
799
[ "`wc -c < /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small/foo.wasm`" -lt "1024" ]
LD_LIBRARY_PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small:/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib:/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-bootstrap-tools/x86_64-unknown-linux-gnu/release/deps:/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/lib" '/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc' --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small -L /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small  foo.rs -C lto -O --target wasm32-unknown-unknown --cfg b
wc -c < /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small/foo.wasm
15814
[ "`wc -c < /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small/foo.wasm`" -lt "5120" ]
------------------------------------------
stderr:
------------------------------------------
------------------------------------------
make: *** [Makefile:11: all] Error 1
------------------------------------------




failures:
    [run-make] run-make/wasm-panic-small

test result: FAILED. 18 passed; 1 failed; 6 ignored; 0 measured; 0 filtered out; finished in 4.01s



command did not execute successfully: "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib/rustlib/wasm32-unknown-unknown/lib" "--rustc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc" "--rustdoc-path" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustdoc" "--src-base" "/checkout/src/test/run-make" "--build-base" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make" "--stage-id" "stage2-wasm32-unknown-unknown" "--suite" "run-make" "--mode" "run-make" "--target" "wasm32-unknown-unknown" "--host" "x86_64-unknown-linux-gnu" "--llvm-filecheck" "/checkout/obj/build/x86_64-unknown-linux-gnu/llvm/build/bin/FileCheck" "--nodejs" "/node-v15.14.0-linux-x64/bin/node" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/x86_64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/wasm32-unknown-unknown/native/rust-test-helpers" "--docck-python" "/usr/bin/python3" "--lldb-python" "/usr/bin/python3" "--gdb" "/usr/bin/gdb" "--llvm-version" "12.0.0-rust-1.53.0-nightly" "--llvm-components" "aarch64 aarch64asmparser aarch64codegen aarch64desc aarch64disassembler aarch64info aarch64utils aggressiveinstcombine all all-targets analysis arm armasmparser armcodegen armdesc armdisassembler arminfo armutils asmparser asmprinter avr avrasmparser avrcodegen avrdesc avrdisassembler avrinfo binaryformat bitreader bitstreamreader bitwriter cfguard codegen core coroutines coverage debuginfocodeview debuginfodwarf debuginfogsym debuginfomsf debuginfopdb demangle dlltooldriver dwarflinker engine executionengine extensions filecheck frontendopenacc frontendopenmp fuzzmutate globalisel hellonew hexagon hexagonasmparser hexagoncodegen hexagondesc hexagondisassembler hexagoninfo instcombine instrumentation interfacestub interpreter ipo irreader jitlink libdriver lineeditor linker lto mc mca mcdisassembler mcjit mcparser mips mipsasmparser mipscodegen mipsdesc mipsdisassembler mipsinfo mirparser msp430 msp430asmparser msp430codegen msp430desc msp430disassembler msp430info native nativecodegen nvptx nvptxcodegen nvptxdesc nvptxinfo objcarcopts object objectyaml option orcjit orcshared orctargetprocess passes powerpc powerpcasmparser powerpccodegen powerpcdesc powerpcdisassembler powerpcinfo profiledata remarks riscv riscvasmparser riscvcodegen riscvdesc riscvdisassembler riscvinfo runtimedyld scalaropts selectiondag sparc sparcasmparser sparccodegen sparcdesc sparcdisassembler sparcinfo support symbolize systemz systemzasmparser systemzcodegen systemzdesc systemzdisassembler systemzinfo tablegen target textapi transformutils vectorize webassembly webassemblyasmparser webassemblycodegen webassemblydesc webassemblydisassembler webassemblyinfo windowsmanifest x86 x86asmparser x86codegen x86desc x86disassembler x86info xray" "--llvm-bin-dir" "/checkout/obj/build/x86_64-unknown-linux-gnu/llvm/build/bin" "--cc" "clang" "--cxx" "clang" "--cflags" "-ffunction-sections -fdata-sections -fPIC --target=wasm32-unknown-unknown" "--ar" "ar" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"


failed to run: /checkout/obj/build/bootstrap/debug/bootstrap --stage 2 test --host= --target wasm32-unknown-unknown src/test/run-make src/test/ui src/test/mir-opt src/test/codegen-units library/core
Build completed unsuccessfully in 0:16:36
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/runs/2314533711)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/runs/2314533711","builder_name":"checks-actions"} -->

nikic commented:
The NewPM fat LTO pipeline is known to be rather weak in LLVM 12 (https://reviews.llvm.org/D96780). I've adjusted the patch to only enable NewPM if fat LTO is not used (it is not by default). Not sure that's a good idea though, as it means we use different pass managers depending on your LTO setting.

nikic commented:
I've collected a profile for the rustc_resolve compile-time regression, and the main thing I noticed there is that it stress-tests loop optimization. It seems to contain a large loop with many exits. I think https://github.com/llvm/llvm-project/commit/8de2f1ff79aa2a8a72b8c44ea70b48aa9f282ae5 may help it, and https://reviews.llvm.org/D100264, and there's another SCEV invalidation issue that needs to be resolved.

nikic commented:
@nagisa Any thoughts on the bit about disabling NewPM for fat LTO? Is that okay to do, or would it be better to delay the switch until LLVM 13 and just land the preparatory parts now?

nagisa commented:
I'm not sure on what's the best way forward is here, either.

I don't particularly love the idea of maintaining two distinct pass managers for an extended period of time. On the other hand the old PM code hasn't changed in any significant way for _ages_ and  its not like it was a source of issues (other than the fact that we fail to optimize one case or another). Meanwhile if we adopted this scheme, the NewPM would still be exercised by a huge amount of code still, as fat LTO is not _that_ heavily used (by the virtue of not being enabled by default).

I guess the question then is: are the gains from NewPM meaningful enough to take on the slight risk here? I'm inclined to think that they may be. 

I guess at this point it might be a good idea to poke @cuviper and see if they have an opinion either way.

cuviper commented:
> Clang has switched the default to NewPM shortly after the LLVM 12 branch.

Meaning it will only be the default in 13, right? If so, I'm not sure we want to be the early adopters on that. Maybe for testing, we can configure the default such that it's enabled in alt builds for now?

Consider this only a weak objection though. If you feel confident in it, I'm game -- and we can always flip back if needed.

> I don't particularly love the idea of maintaining two distinct pass managers for an extended period of time.

Note that if we have to keep the old PM code around for external LLVM anyway, until 13 is our minimum if that's where we commit fully ourselves. The only way we avoid a transition like that is if we wait for the minimum to support everything, but I doubt you want that. So let's just accept that we'll need both PMs for some time.

nagisa commented:
Yeah makes sense to me. If clang is not quite enabling NewPM before 13 yet, we shouldn't either.

nikic commented:
@nagisa Okay, I've dropped the part that actually enabled NewPM now, let's delay that to LLVM 13.

nagisa commented:
@bors r+ rollup=never

bors commented:
:pushpin: Commit 8d4bdf565c1af1dc18abc40a236bdda62f1c96ad has been approved by `nagisa`

<!-- @bors r=nagisa 8d4bdf565c1af1dc18abc40a236bdda62f1c96ad -->
<!-- homu: {"type":"Approved","sha":"8d4bdf565c1af1dc18abc40a236bdda62f1c96ad","approver":"nagisa"} -->

bors commented:
:hourglass: Testing commit 8d4bdf565c1af1dc18abc40a236bdda62f1c96ad with merge 5157e081399fc5986f18bd5fdd9081ed8a4993fa...
<!-- homu: {"type":"BuildStarted","head_sha":"8d4bdf565c1af1dc18abc40a236bdda62f1c96ad","merge_sha":"5157e081399fc5986f18bd5fdd9081ed8a4993fa"} -->

rust-log-analyzer commented:
The job **`x86_64-msvc-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/2443071309) [(plain)](https://github.com/rust-lang-ci/rust/commit/5157e081399fc5986f18bd5fdd9081ed8a4993fa/checks/2443071309/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
---- [ui] ui\backtrace-debuginfo.rs stdout ----

error: test run failed!
status: exit code: 101
command: PATH="D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage2\lib\rustlib\x86_64-pc-windows-msvc\lib;C:\Program Files (x86)\Windows Kits\10\bin\x64;C:\Program Files (x86)\Windows Kits\10\bin\10.0.19041.0\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\bin\HostX64\x64;C:\Program Files (x86)\Microsoft Visual Studio\2019\Enterprise\VC\Tools\MSVC\14.28.29910\bin\HostX64\x64;D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0-bootstrap-tools\x86_64-pc-windows-msvc\release\deps;D:\a\rust\rust\build\x86_64-pc-windows-msvc\stage0\bin;D:\a\rust\rust\ninja;D:\a\rust\rust\msys2\mingw64\bin;C:\hostedtoolcache\windows\Python\3.9.4\x64\Scripts;C:\hostedtoolcache\windows\Python\3.9.4\x64;C:\msys64\usr\bin;D:\a\rust\rust\sccache;C:\Users\runneradmin\.dotnet\tools;C:\Program Files\MongoDB\Server\4.4\bin;C:\aliyun-cli;C:\vcpkg;C:\cf-cli;C:\Program Files (x86)\NSIS;C:\tools\zstd;C:\Program Files\Mercurial;C:\hostedtoolcache\windows\stack\2.5.1\x64;C:\tools\ghc-9.0.1\bin;C:\Program Files\dotnet;C:\mysql-5.7.21-winx64\bin;C:\Program Files\R\R-4.0.5\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\Program Files (x86)\sbt\bin;C:\Rust\.cargo\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\bin;C:\Program Files (x86)\pipx_bin;C:\hostedtoolcache\windows\go\1.15.11\x64\bin;C:\hostedtoolcache\windows\Python\3.7.9\x64\Scripts;C:\hostedtoolcache\windows\Python\3.7.9\x64;C:\hostedtoolcache\windows\Ruby\2.5.8\x64\bin;C:\hostedtoolcache\windows\Java_Adopt_jdk\8.0.282-8\x64\bin;C:\npm\prefix;C:\Program Files\Microsoft SDKs\Azure\Azure Dev Spaces CLI;C:\Program Files\Microsoft SDKs\Azure\Azure Dev Spaces CLI;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\ProgramData\Chocolatey\bin;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files\Docker;C:\Program Files\PowerShell\7;C:\Program Files\dotnet;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn;C:\Program Files\nodejs;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.6.3\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\OpenSSL\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\tools\php;C:\Program Files (x86)\sbt\bin;C:\Program Files\TortoiseSVN\bin;C:\SeleniumWebDrivers\ChromeDriver;C:\SeleniumWebDrivers\EdgeDriver;C:\Program Files\CMake\bin;C:\Program Files\Amazon\AWSCLIV2;C:\Program Files\Amazon\SessionManagerPlugin\bin;C:\Program Files\Amazon\AWSSAMCLI\bin;C:\Program Files (x86)\Google\Cloud SDK\google-cloud-sdk\bin;C:\Program Files (x86)\Microsoft BizTalk Server;C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\test\\ui\\backtrace-debuginfo\\a.exe"
------------------------------------------
---------------------------------------
trace does not match position list
trace does not match position list
still need to find ["backtrace-debuginfo.rs:183", "backtrace-debuginfo.rs:119", "backtrace-debuginfo.rs:82", "backtrace-debuginfo-aux.rs:6"]
--- stdout
backtrace-debuginfo-aux.rs:6
backtrace-debuginfo.rs:82
backtrace-debuginfo.rs:119
backtrace-debuginfo.rs:119
backtrace-debuginfo.rs:183

--- stderr
test case 2
thread 'main' panicked at 'explicit panic', D:\a\rust\rust\src/test\ui\backtrace-debuginfo.rs:83:9
stack backtrace:
   0:     0x7ffdcfa8845e - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2acd651982b597b3
   1:     0x7ffdcfab41bd - core::fmt::write::hb03e13d59cf1fd5f
   2:     0x7ffdcfa7be18 - <std::io::IoSlice as core::fmt::Debug>::fmt::hd2fcbf5878964876
   3:     0x7ffdcfa8c8f2 - std::panicking::take_hook::hec0a9630d503a271
   4:     0x7ffdcfa8c36b - std::panicking::take_hook::hec0a9630d503a271
   5:     0x7ffdcfa8d093 - std::panicking::rust_panic_with_hook::h53f2c9d7b3eaa7d2
   6:     0x7ff6374510a8 - std::panicking::begin_panic::{{closure}}<str>
                               at D:\a\rust\rust\library\std\src\panicking.rs:520
   7:     0x7ff637451032 - std::sys_common::backtrace::__rust_end_short_backtrace<closure-0,!>
                               at D:\a\rust\rust\library\std\src\sys_common\backtrace.rs:141
   8:     0x7ff63745581f - std::panicking::begin_panic<str>
                               at D:\a\rust\rust\library\std\src\panicking.rs:519
   9:     0x7ff637452a23 - backtrace_debuginfo::inner::{{closure}}
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:83
  10:     0x7ff637452a23 - backtrace_debuginfo::aux::callback<closure-0>
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo-aux.rs:7
  11:     0x7ff637452f06 - backtrace_debuginfo::inner
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:82
  12:     0x7ff637453280 - backtrace_debuginfo::outer
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:119
  13:     0x7ff637453674 - backtrace_debuginfo::main
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:183
  14:     0x7ff63745104c - core::ops::function::FnOnce::call_once
                               at D:\a\rust\rust\library\core\src\ops\function.rs:227
  15:     0x7ff63745104c - std::sys_common::backtrace::__rust_begin_short_backtrace<fn(),tuple<>>
                               at D:\a\rust\rust\library\std\src\sys_common\backtrace.rs:125
  16:     0x7ff637451072 - std::rt::lang_start::{{closure}}<tuple<>>
                               at D:\a\rust\rust\library\std\src\rt.rs:49
  17:     0x7ffdcfa8d367 - std::rt::lang_start_internal::h2d1ffef100a23ae8
  18:     0x7ff637454b6b - main
  19:     0x7ff637454eb0 - invoke_main
                               at D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  20:     0x7ff637454eb0 - __scrt_common_main_seh
                               at D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  21:     0x7ffe012d7974 - BaseThreadInitThunk
  22:     0x7ffe01b0a2f1 - RtlUserThreadStart
---------------------------------------
trace does not match position list
trace does not match position list
still need to find ["backtrace-debuginfo.rs:183", "backtrace-debuginfo.rs:119", "backtrace-debuginfo.rs:85", "backtrace-debuginfo-aux.rs:13"]
--- stdout
backtrace-debuginfo-aux.rs:13
backtrace-debuginfo.rs:85
backtrace-debuginfo.rs:119
backtrace-debuginfo.rs:119
backtrace-debuginfo.rs:183

--- stderr
test case 3
thread 'main' panicked at 'explicit panic', D:\a\rust\rust\src/test\ui\backtrace-debuginfo.rs:86:9
stack backtrace:
   0:     0x7ffdcfa8845e - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2acd651982b597b3
   1:     0x7ffdcfab41bd - core::fmt::write::hb03e13d59cf1fd5f
   2:     0x7ffdcfa7be18 - <std::io::IoSlice as core::fmt::Debug>::fmt::hd2fcbf5878964876
   3:     0x7ffdcfa8c8f2 - std::panicking::take_hook::hec0a9630d503a271
   4:     0x7ffdcfa8c36b - std::panicking::take_hook::hec0a9630d503a271
   5:     0x7ffdcfa8d093 - std::panicking::rust_panic_with_hook::h53f2c9d7b3eaa7d2
   6:     0x7ff6374510a8 - std::panicking::begin_panic::{{closure}}<str>
                               at D:\a\rust\rust\library\std\src\panicking.rs:520
   7:     0x7ff637451032 - std::sys_common::backtrace::__rust_end_short_backtrace<closure-0,!>
Some tests failed in compiletest suite=ui mode=ui host=x86_64-pc-windows-msvc target=x86_64-pc-windows-msvc
                               at D:\a\rust\rust\library\std\src\sys_common\backtrace.rs:141
   8:     0x7ff63745581f - std::panicking::begin_panic<str>
                               at D:\a\rust\rust\library\std\src\panicking.rs:519
   9:     0x7ff637452ba3 - backtrace_debuginfo::inner::{{closure}}
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:86
  10:     0x7ff637452ba3 - backtrace_debuginfo::aux::callback_inlined<closure-1>
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo-aux.rs:14
  11:     0x7ff637452f36 - backtrace_debuginfo::inner
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:85
  12:     0x7ff637453280 - backtrace_debuginfo::outer
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:119
  13:     0x7ff637453674 - backtrace_debuginfo::main
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:183
  14:     0x7ff63745104c - core::ops::function::FnOnce::call_once
                               at D:\a\rust\rust\library\core\src\ops\function.rs:227
  15:     0x7ff63745104c - std::sys_common::backtrace::__rust_begin_short_backtrace<fn(),tuple<>>
                               at D:\a\rust\rust\library\std\src\sys_common\backtrace.rs:125
  16:     0x7ff637451072 - std::rt::lang_start::{{closure}}<tuple<>>
                               at D:\a\rust\rust\library\std\src\rt.rs:49
  17:     0x7ffdcfa8d367 - std::rt::lang_start_internal::h2d1ffef100a23ae8
  18:     0x7ff637454b6b - main
  19:     0x7ff637454eb0 - invoke_main
                               at D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  20:     0x7ff637454eb0 - __scrt_common_main_seh
                               at D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  21:     0x7ffe012d7974 - BaseThreadInitThunk
  22:     0x7ffe01b0a2f1 - RtlUserThreadStart
---------------------------------------
trace does not match position list
trace does not match position list
still need to find ["backtrace-debuginfo.rs:183", "backtrace-debuginfo.rs:120", "backtrace-debuginfo.rs:105", "backtrace-debuginfo-aux.rs:6"]
--- stdout
backtrace-debuginfo-aux.rs:6
backtrace-debuginfo.rs:105
backtrace-debuginfo.rs:120
backtrace-debuginfo.rs:120
backtrace-debuginfo.rs:183

--- stderr
test case 7
thread 'main' panicked at 'explicit panic', D:\a\rust\rust\src/test\ui\backtrace-debuginfo.rs:106:9
stack backtrace:
   0:     0x7ffdcfa8845e - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2acd651982b597b3
   1:     0x7ffdcfab41bd - core::fmt::write::hb03e13d59cf1fd5f
   2:     0x7ffdcfa7be18 - <std::io::IoSlice as core::fmt::Debug>::fmt::hd2fcbf5878964876
   3:     0x7ffdcfa8c8f2 - std::panicking::take_hook::hec0a9630d503a271
   4:     0x7ffdcfa8c36b - std::panicking::take_hook::hec0a9630d503a271
   5:     0x7ffdcfa8d093 - std::panicking::rust_panic_with_hook::h53f2c9d7b3eaa7d2
   6:     0x7ff6374510a8 - std::panicking::begin_panic::{{closure}}<str>
                               at D:\a\rust\rust\library\std\src\panicking.rs:520
   7:     0x7ff637451032 - std::sys_common::backtrace::__rust_end_short_backtrace<closure-0,!>
                               at D:\a\rust\rust\library\std\src\sys_common\backtrace.rs:141
   8:     0x7ff63745581f - std::panicking::begin_panic<str>
                               at D:\a\rust\rust\library\std\src\panicking.rs:519
   9:     0x7ff637452ae3 - backtrace_debuginfo::inner_inlined::{{closure}}
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:106
  10:     0x7ff637452ae3 - backtrace_debuginfo::aux::callback<closure-0>
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo-aux.rs:7
  11:     0x7ff637453098 - backtrace_debuginfo::inner_inlined
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:105
  12:     0x7ff6374532b2 - backtrace_debuginfo::outer
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:120
  13:     0x7ff637453674 - backtrace_debuginfo::main
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:183
  14:     0x7ff63745104c - core::ops::function::FnOnce::call_once
                               at D:\a\rust\rust\library\core\src\ops\function.rs:227
  15:     0x7ff63745104c - std::sys_common::backtrace::__rust_begin_short_backtrace<fn(),tuple<>>
                               at D:\a\rust\rust\library\std\src\sys_common\backtrace.rs:125
  16:     0x7ff637451072 - std::rt::lang_start::{{closure}}<tuple<>>
                               at D:\a\rust\rust\library\std\src\rt.rs:49
  17:     0x7ffdcfa8d367 - std::rt::lang_start_internal::h2d1ffef100a23ae8
  18:     0x7ff637454b6b - main
  19:     0x7ff637454eb0 - invoke_main
                               at D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  20:     0x7ff637454eb0 - __scrt_common_main_seh
                               at D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  21:     0x7ffe012d7974 - BaseThreadInitThunk
  22:     0x7ffe01b0a2f1 - RtlUserThreadStart
---------------------------------------
trace does not match position list
trace does not match position list
still need to find ["backtrace-debuginfo.rs:183", "backtrace-debuginfo.rs:120", "backtrace-debuginfo.rs:108", "backtrace-debuginfo-aux.rs:13"]
--- stdout
backtrace-debuginfo-aux.rs:13
backtrace-debuginfo.rs:108
backtrace-debuginfo.rs:120
backtrace-debuginfo.rs:120
backtrace-debuginfo.rs:183

--- stderr
test case 8
thread 'main' panicked at 'explicit panic', D:\a\rust\rust\src/test\ui\backtrace-debuginfo.rs:109:9
stack backtrace:
   0:     0x7ffdcfa8845e - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h2acd651982b597b3
   1:     0x7ffdcfab41bd - core::fmt::write::hb03e13d59cf1fd5f
   2:     0x7ffdcfa7be18 - <std::io::IoSlice as core::fmt::Debug>::fmt::hd2fcbf5878964876
   3:     0x7ffdcfa8c8f2 - std::panicking::take_hook::hec0a9630d503a271
   4:     0x7ffdcfa8c36b - std::panicking::take_hook::hec0a9630d503a271
   5:     0x7ffdcfa8d093 - std::panicking::rust_panic_with_hook::h53f2c9d7b3eaa7d2
   6:     0x7ff6374510a8 - std::panicking::begin_panic::{{closure}}<str>
                               at D:\a\rust\rust\library\std\src\panicking.rs:520
   7:     0x7ff637451032 - std::sys_common::backtrace::__rust_end_short_backtrace<closure-0,!>
                               at D:\a\rust\rust\library\std\src\sys_common\backtrace.rs:141
   8:     0x7ff63745581f - std::panicking::begin_panic<str>
                               at D:\a\rust\rust\library\std\src\panicking.rs:519
   9:     0x7ff637452c63 - backtrace_debuginfo::inner_inlined::{{closure}}
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:109
  10:     0x7ff637452c63 - backtrace_debuginfo::aux::callback_inlined<closure-1>
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo-aux.rs:14
  11:     0x7ff6374530c8 - backtrace_debuginfo::inner_inlined
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:108
  12:     0x7ff6374532b2 - backtrace_debuginfo::outer
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:120
  13:     0x7ff637453674 - backtrace_debuginfo::main
                               at D:\a\rust\rust\src\test\ui\backtrace-debuginfo.rs:183
  14:     0x7ff63745104c - core::ops::function::FnOnce::call_once
                               at D:\a\rust\rust\library\core\src\ops\function.rs:227
  15:     0x7ff63745104c - std::sys_common::backtrace::__rust_begin_short_backtrace<fn(),tuple<>>
                               at D:\a\rust\rust\library\std\src\sys_common\backtrace.rs:125
  16:     0x7ff637451072 - std::rt::lang_start::{{closure}}<tuple<>>
                               at D:\a\rust\rust\library\std\src\rt.rs:49
  17:     0x7ffdcfa8d367 - std::rt::lang_start_internal::h2d1ffef100a23ae8
  18:     0x7ff637454b6b - main
  19:     0x7ff637454eb0 - invoke_main
                               at D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:78
  20:     0x7ff637454eb0 - __scrt_common_main_seh
                               at D:\a01\_work\26\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
  21:     0x7ffe012d7974 - BaseThreadInitThunk
  22:     0x7ffe01b0a2f1 - RtlUserThreadStart

------------------------------------------
stderr:
------------------------------------------
------------------------------------------
thread 'main' panicked at 'found some errors', D:\a\rust\rust\src/test\ui\backtrace-debuginfo.rs:173:9

------------------------------------------


---
test result: FAILED. 11670 passed; 1 failed; 132 ignored; 0 measured; 0 filtered out; finished in 368.22s



command did not execute successfully: "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage0-tools-bin\\compiletest.exe" "--compile-lib-path" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\bin" "--run-lib-path" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\lib\\rustlib\\x86_64-pc-windows-msvc\\lib" "--rustc-path" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\stage2\\bin\\rustc.exe" "--src-base" "D:\\a\\rust\\rust\\src/test\\ui" "--build-base" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\test\\ui" "--stage-id" "stage2-x86_64-pc-windows-msvc" "--suite" "ui" "--mode" "ui" "--target" "x86_64-pc-windows-msvc" "--host" "x86_64-pc-windows-msvc" "--llvm-filecheck" "D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\llvm\\build\\bin\\FileCheck.exe" "--nodejs" "C:\\Program Files\\nodejs\\node" "--npm" "C:\\Program Files\\nodejs\\npm" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\native\\rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=D:\\a\\rust\\rust\\build\\x86_64-pc-windows-msvc\\native\\rust-test-helpers" "--docck-python" "C:\\hostedtoolcache\\windows\\Python\\3.9.4\\x64\\python3.exe" "--lldb-python" "C:\\hostedtoolcache\\windows\\Python\\3.9.4\\x64\\python3.exe" "--gdb" "C:\\msys64\\usr\\bin\\gdb" "--llvm-version" "12.0.0-rust-1.53.0-nightly" "--llvm-components" "aarch64 aarch64asmparser aarch64codegen aarch64desc aarch64disassembler aarch64info aarch64utils aggressiveinstcombine all all-targets analysis arm armasmparser armcodegen armdesc armdisassembler arminfo armutils asmparser asmprinter avr avrasmparser avrcodegen avrdesc avrdisassembler avrinfo binaryformat bitreader bitstreamreader bitwriter cfguard codegen core coroutines coverage debuginfocodeview debuginfodwarf debuginfogsym debuginfomsf debuginfopdb demangle dlltooldriver dwarflinker engine executionengine extensions filecheck frontendopenacc frontendopenmp fuzzmutate globalisel hellonew hexagon hexagonasmparser hexagoncodegen hexagondesc hexagondisassembler hexagoninfo instcombine instrumentation interfacestub interpreter ipo irreader jitlink libdriver lineeditor linker lto mc mca mcdisassembler mcjit mcparser mips mipsasmparser mipscodegen mipsdesc mipsdisassembler mipsinfo mirparser msp430 msp430asmparser msp430codegen msp430desc msp430disassembler msp430info native nativecodegen nvptx nvptxcodegen nvptxdesc nvptxinfo objcarcopts object objectyaml option orcjit orcshared orctargetprocess passes powerpc powerpcasmparser powerpccodegen powerpcdesc powerpcdisassembler powerpcinfo profiledata remarks riscv riscvasmparser riscvcodegen riscvdesc riscvdisassembler riscvinfo runtimedyld scalaropts selectiondag sparc sparcasmparser sparccodegen sparcdesc sparcdisassembler sparcinfo support symbolize systemz systemzasmparser systemzcodegen systemzdesc systemzdisassembler systemzinfo tablegen target textapi transformutils vectorize webassembly webassemblyasmparser webassemblycodegen webassemblydesc webassemblydisassembler webassemblyinfo windowsmanifest x86 x86asmparser x86codegen x86desc x86disassembler x86info xray" "--cc" "" "--cxx" "" "--cflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"


failed to run: D:\a\rust\rust\build\bootstrap\debug\bootstrap test --stage 2 src/test/ui src/tools/linkchecker
Build completed unsuccessfully in 0:30:18
Build completed unsuccessfully in 0:30:18
make: *** [Makefile:78: ci-subset-2] Error 1
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/runs/2443264178)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/runs/2443264178","builder_name":"checks-actions"} -->

nikic commented:
Added `-C opt-level=0` to that test, hopefully that's sufficient.

@bors r=nagisa

bors commented:
:pushpin: Commit fdacbd2574fb891e2ba1a19615ae6ed2dac26b53 has been approved by `nagisa`

<!-- @bors r=nagisa fdacbd2574fb891e2ba1a19615ae6ed2dac26b53 -->
<!-- homu: {"type":"Approved","sha":"fdacbd2574fb891e2ba1a19615ae6ed2dac26b53","approver":"nagisa"} -->

cuviper commented:
FYI, a few days ago Arthur proposed legacy PM deprecation:

https://lists.llvm.org/pipermail/llvm-dev/2021-April/150100.html
> We should have a deprecation timeline that revolves around the major releases. I'd say we announce the legacy PM for the optimization pipeline to be deprecated the major release after the new PM has been the default. Looks like the new PM switch made it into LLVM 12.

He was corrected that the switch will only happen in 13, so that would mean it could be announced deprecated in 14, and I suppose it may start getting crippled or even removed soon after. Regardless, that's more emphasis that we should definitely switch rustc when we get to 13.

bors commented:
:hourglass: Testing commit fdacbd2574fb891e2ba1a19615ae6ed2dac26b53 with merge 028fe2358ac144ac35144a9e2e31dfe00a0cbdc0...
<!-- homu: {"type":"BuildStarted","head_sha":"fdacbd2574fb891e2ba1a19615ae6ed2dac26b53","merge_sha":"028fe2358ac144ac35144a9e2e31dfe00a0cbdc0"} -->

rust-log-analyzer commented:
The job **`aarch64-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/runs/2453469272) [(plain)](https://github.com/rust-lang-ci/rust/commit/028fe2358ac144ac35144a9e2e31dfe00a0cbdc0/checks/2453469272/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
failures:

---- [ui] ui/extern/issue-64655-extern-rust-must-allow-unwind.rs#fat0 stdout ----

error in revision `fat0`: test compilation failed although it shouldn't!
status: signal: 6 (core dumped)
command: "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/extern/issue-64655-extern-rust-must-allow-unwind.rs" "-Zthreads=1" "--target=aarch64-unknown-linux-gnu" "--cfg" "fat0" "--error-format" "json" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zemit-future-incompat-report" "-o" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/extern/issue-64655-extern-rust-must-allow-unwind.fat0/a" "-Crpath" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "-C" "opt-level=0" "-C" "lto=fat" "-L" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/extern/issue-64655-extern-rust-must-allow-unwind.fat0/auxiliary"
------------------------------------------

------------------------------------------
stderr:
stderr:
------------------------------------------
rustc: /checkout/src/llvm-project/llvm/include/llvm/ADT/APInt.h:1634: uint64_t llvm::APInt::getZExtValue() const: Assertion `getActiveBits() <= 64 && "Too many bits for uint64_t"' failed.
------------------------------------------


---- [ui] ui/sanitize/new-llvm-pass-manager-thin-lto.rs#opt0 stdout ----
---- [ui] ui/sanitize/new-llvm-pass-manager-thin-lto.rs#opt0 stdout ----

error in revision `opt0`: test compilation failed although it shouldn't!
status: signal: 6 (core dumped)
command: "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "/checkout/src/test/ui/sanitize/new-llvm-pass-manager-thin-lto.rs" "-Zthreads=1" "--target=aarch64-unknown-linux-gnu" "--cfg" "opt0" "--error-format" "json" "-Ccodegen-units=1" "-Zui-testing" "-Zdeduplicate-diagnostics=no" "-Zemit-future-incompat-report" "-o" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/a" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Zunstable-options" "-Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "-Znew-llvm-pass-manager=yes" "-Zsanitizer=address" "-Clto=thin" "-Copt-level=0" "-L" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui/sanitize/new-llvm-pass-manager-thin-lto.opt0/auxiliary"
------------------------------------------

------------------------------------------
stderr:
stderr:
------------------------------------------
rustc: /checkout/src/llvm-project/llvm/include/llvm/ADT/APInt.h:1634: uint64_t llvm::APInt::getZExtValue() const: Assertion `getActiveBits() <= 64 && "Too many bits for uint64_t"' failed.
------------------------------------------



---

Some tests failed in compiletest suite=ui mode=ui host=aarch64-unknown-linux-gnu target=aarch64-unknown-linux-gnu


command did not execute successfully: "/checkout/obj/build/aarch64-unknown-linux-gnu/stage0-tools-bin/compiletest" "--compile-lib-path" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/lib" "--run-lib-path" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/lib/rustlib/aarch64-unknown-linux-gnu/lib" "--rustc-path" "/checkout/obj/build/aarch64-unknown-linux-gnu/stage2/bin/rustc" "--src-base" "/checkout/src/test/ui" "--build-base" "/checkout/obj/build/aarch64-unknown-linux-gnu/test/ui" "--stage-id" "stage2-aarch64-unknown-linux-gnu" "--suite" "ui" "--mode" "ui" "--target" "aarch64-unknown-linux-gnu" "--host" "aarch64-unknown-linux-gnu" "--llvm-filecheck" "/checkout/obj/build/aarch64-unknown-linux-gnu/llvm/build/bin/FileCheck" "--host-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "--target-rustcflags" "-Crpath -O -Cdebuginfo=0 -Zunstable-options  -Lnative=/checkout/obj/build/aarch64-unknown-linux-gnu/native/rust-test-helpers" "--docck-python" "/usr/bin/python3" "--lldb-python" "/usr/bin/python3" "--gdb" "/usr/bin/gdb" "--llvm-version" "12.0.0-rust-1.53.0-nightly" "--llvm-components" "aarch64 aarch64asmparser aarch64codegen aarch64desc aarch64disassembler aarch64info aarch64utils aggressiveinstcombine all all-targets analysis arm armasmparser armcodegen armdesc armdisassembler arminfo armutils asmparser asmprinter avr avrasmparser avrcodegen avrdesc avrdisassembler avrinfo binaryformat bitreader bitstreamreader bitwriter cfguard codegen core coroutines coverage debuginfocodeview debuginfodwarf debuginfogsym debuginfomsf debuginfopdb demangle dlltooldriver dwarflinker engine executionengine extensions filecheck frontendopenacc frontendopenmp fuzzmutate globalisel hellonew hexagon hexagonasmparser hexagoncodegen hexagondesc hexagondisassembler hexagoninfo instcombine instrumentation interfacestub interpreter ipo irreader jitlink libdriver lineeditor linker lto mc mca mcdisassembler mcjit mcparser mips mipsasmparser mipscodegen mipsdesc mipsdisassembler mipsinfo mirparser msp430 msp430asmparser msp430codegen msp430desc msp430disassembler msp430info native nativecodegen nvptx nvptxcodegen nvptxdesc nvptxinfo objcarcopts object objectyaml option orcjit orcshared orctargetprocess passes powerpc powerpcasmparser powerpccodegen powerpcdesc powerpcdisassembler powerpcinfo profiledata remarks riscv riscvasmparser riscvcodegen riscvdesc riscvdisassembler riscvinfo runtimedyld scalaropts selectiondag sparc sparcasmparser sparccodegen sparcdesc sparcdisassembler sparcinfo support symbolize systemz systemzasmparser systemzcodegen systemzdesc systemzdisassembler systemzinfo tablegen target textapi transformutils vectorize webassembly webassemblyasmparser webassemblycodegen webassemblydesc webassemblydisassembler webassemblyinfo windowsmanifest x86 x86asmparser x86codegen x86desc x86disassembler x86info xray" "--cc" "" "--cxx" "" "--cflags" "" "--adb-path" "adb" "--adb-test-dir" "/data/tmp/work" "--android-cross-path" "" "--color" "always"


failed to run: /checkout/obj/build/bootstrap/debug/bootstrap --stage 2 test
Build completed unsuccessfully in 0:47:30
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/runs/2453747468)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/runs/2453747468","builder_name":"checks-actions"} -->

nikic commented:
Reduction of the assertion failure:
```
; RUN: llc -O0
target triple = "aarch64-unknown-linux-gnu"

define void @test() {
  switch i128 0, label %label1 [
    i128 1329227995784915872903807060280344576, label %label1
  ]

label1:
  ret void
}
```
Which is already fixed by https://github.com/llvm/llvm-project/commit/c2b322fc19e829162ed4c7dcd04d9e9b2cd4e66c and backported in https://github.com/llvm/llvm-project/commit/fa0971b87fb2c9d14d1bba2551e61f02f18f329b -- and which is the first 12.x commit *not* in our fork, duh.

nikic committed:
Update LLVM submodule

This updates the LLVM submodule with recent LLVM 12.x fixes. In
particular, it resolves an assertion failure when targeting
AArch64 at O0.

nikic commented:
I've merged current upstream 12.x into our fork, and have added the submodule update here. That should resolve the AArch64 assertion failure.

nagisa commented:
@bors r+

bors commented:
:pushpin: Commit 1b928ff8f8b069f45fde420dabf34d9c53b9cb3a has been approved by `nagisa`

<!-- @bors r=nagisa 1b928ff8f8b069f45fde420dabf34d9c53b9cb3a -->
<!-- homu: {"type":"Approved","sha":"1b928ff8f8b069f45fde420dabf34d9c53b9cb3a","approver":"nagisa"} -->

bors commented:
:hourglass: Testing commit 1b928ff8f8b069f45fde420dabf34d9c53b9cb3a with merge 7a2f4468892a9bf694b844f1fa1032779320c7e5...
<!-- homu: {"type":"BuildStarted","head_sha":"1b928ff8f8b069f45fde420dabf34d9c53b9cb3a","merge_sha":"7a2f4468892a9bf694b844f1fa1032779320c7e5"} -->

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/runs/2540125382)
Approved by: nagisa
Pushing 7a2f4468892a9bf694b844f1fa1032779320c7e5 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"nagisa","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/runs/2540125382"},"merge_sha":"7a2f4468892a9bf694b844f1fa1032779320c7e5"} -->

nikic commented:
Hrm, it looks like the `wasm-panic-small` test still fails even with the LLVM 13 upgrade:

```
LD_LIBRARY_PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small:/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib:/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-bootstrap-tools/x86_64-unknown-linux-gnu/release/deps:/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/lib" '/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc' --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small -L /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small  foo.rs -C lto -O --target wasm32-unknown-unknown --cfg a
wc -c < /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small/foo.wasm
805
[ "`wc -c < /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small/foo.wasm`" -lt "1024" ]
LD_LIBRARY_PATH="/checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small:/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/lib:/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-bootstrap-tools/x86_64-unknown-linux-gnu/release/deps:/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/lib" '/checkout/obj/build/x86_64-unknown-linux-gnu/stage2/bin/rustc' --out-dir /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small -L /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small  foo.rs -C lto -O --target wasm32-unknown-unknown --cfg b
wc -c < /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small/foo.wasm
15687
[ "`wc -c < /checkout/obj/build/x86_64-unknown-linux-gnu/test/run-make/wasm-panic-small/wasm-panic-small/foo.wasm`" -lt "5120" ]
```
